### PR TITLE
feat: prepend use-client directive for with Next.js App Router

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Support app router of Nextjs. |
| 🇨🇳 Chinese | 支持 Nextjs app router。 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at a451d95</samp>

This pull request fixes various TypeScript errors and code style issues in several components of the `ant-design` library. It also refactors the `Affix` component to support server-side rendering (SSR) and adds a `no-default-export` rule exception to some component files.

### <samp>🤖 Generated by Copilot at a451d95</samp>

*  Add `/* eslint-disable @typescript-eslint/no-unused-vars */` directive to 20 component files ([link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-18949a009c0a2f7f47a0136b5808eade7a157b3d78661edc0f75463594e2ad5dR1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-ea17437115f696bb5e9eda1c71130b498ea13d5aee96f5d40bce255e8c68b8e7R1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-bce8b84c85cec197fd50b0e18482c4c8d0d8a05eb51d9714b869f4d9f0abe6caR1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-95b0dcfbdd9a195d1d0141272a065d901452a60a60632bd0e30f74e80eb85a6bR1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-c55203058d3a4064d8f52a5b4e7d7770153d215966d276d6814011d99268f76dL1-R1), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-eafdeaa24b5e2771367e4ea1383becc626c63072ccd786648d000fa1df282fcaR1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-22587f2ed95b1c480472e426d68bdb6cc32d38b3134e9aa5a32101b973312618R1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-969bd9a7c5b9c1cd26127d496c4a1e4181d847a2000b9b5f272e4714007b6391R1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-da5ca906e2a2a35f0eeeacfcbd4cf2bc6288817ddd292e5ce858f150623606d6R1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-e17ebc19f2ad32b347a03e08a1b4f367a2280027a3e42d2ee4e4c737a57a2243R1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-013735da2a9e27cd090e2a7b89864881fcab5e33ff74b538079f42b82ccd3d24R1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-8a8c225c4369c726cc4f6f3c5dbe979d57e69d5d76a3a0d04fe56016d5aa2c54R1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-1158da0914c171fa74b4fa496bef7272ca0776317fb5e222bebdf11477b191fbR1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-4737f6ef564486c449ec019f2826176d8b8d186088e5e912ad06ad1919b2d804R1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-8220e5af92084a379caeaf225cdad0f0090e0858855f703909a9376e2c037026R1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-6ffd57ad327ed714187e4e46d7289dfe4b84c6e51b08535c56115f3b2847143bR1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-2d2d4fb9566406cae122a321d0c5bb8e9006d00b7a7804bc408ba0ad5f81c098R1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-937e920c7edf3c1b09cde8ed592d2ccc37d8f6eaee8074582eec32dd70c90850L1-R6), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-0a9895a32672ec55a084aa7165fc8666008942903b8e51e30b662e9e1ecffb56R1-R2), [link](https://github.com/ant-design/ant-design/pull/43573/files?diff=unified&w=0#diff-78eb7aa1662e48dc14665cbcead280795a60503742024dd031b87311186be5eeR1-R2)) as part of a refactor to support SSR

### CODE REVIEW

File: components/affix/index.tsx [hunk 1]
```diff
@@ -286,7 +288,7 @@ class InternalAffix extends React.Component<InternalAffixProps, AffixState> {
     return (
       <ResizeObserver onResize={this.updatePosition}>
         <div {...props} ref={this.placeholderNodeRef}>
-          {affixStyle && <div style={placeholderStyle} aria-hidden="true" />}
+          {affixStyle && <div style={placeholderStyle} aria-hidden='true' />}
```
Comment (MadCcc) [inline components/affix/index.tsx:291]:
```suggestion
          {affixStyle && <div style={placeholderStyle} aria-hidden="true" />}
```

File: components/badge/index.tsx [hunk 2]
```diff
@@ -259,7 +261,7 @@ const InternalBadge: React.ForwardRefRenderFunction<HTMLSpanElement, BadgeProps>
               count={displayCount}
               title={titleNode}
               style={scrollNumberStyle}
-              key="scrollNumber"
+              key='scrollNumber'
```
Comment (MadCcc) [inline components/badge/index.tsx:264]:
```suggestion
              key="scrollNumber"
```

File: components/divider/index.tsx [hunk 3]
```diff
@@ -88,7 +90,7 @@ const Divider: React.FC<DividerProps> = (props) => {
       className={classString}
       style={{ ...divider?.style, ...style }}
       {...restProps}
-      role="separator"
+      role='separator'
```
Comment (MadCcc) [inline components/divider/index.tsx:93]:
```suggestion
      role="separator"
```

**Comment by MadCcc** [comment]
html tag 都用双引号，要还原下

**Comment by li-jia-nan** [comment]
> html tag 都用双引号，要还原下

OK

File: components/affix/index.tsx (hunk 1 above)
Comment (zombieJ) [reply components/affix/index.tsx]: prettier 又搞事情？

File: components/cascader/index.tsx [hunk 4]
```diff
@@ -196,7 +198,7 @@ const Cascader = React.forwardRef((props: CascaderProps<any>, ref: React.Ref<Cas
 
   // =================== No Found ====================
   const mergedNotFoundContent = notFoundContent || renderEmpty?.('Cascader') || (
-    <DefaultRenderEmpty componentName="Cascader" />
+    <DefaultRenderEmpty componentName='Cascader' />
```
Comment (li-jia-nan) [inline components/cascader/index.tsx:201]:
```suggestion
    <DefaultRenderEmpty componentName="Cascader" />
```

File: components/list/index.tsx [hunk 5]
```diff
@@ -268,7 +270,7 @@ function List<T>({
     childrenContent = (
       <div className={`${prefixCls}-empty-text`}>
         {(locale && locale.emptyText) || renderEmpty?.('List') || (
-          <DefaultRenderEmpty componentName="List" />
+          <DefaultRenderEmpty componentName='List' />
```
Comment (li-jia-nan) [inline components/list/index.tsx:273]:
```suggestion
          <DefaultRenderEmpty componentName="List" />
```

File: components/mentions/index.tsx [hunk 6]
```diff
@@ -127,14 +129,14 @@ const InternalMentions: React.ForwardRefRenderFunction<MentionsRef, MentionProps
     if (notFoundContent !== undefined) {
       return notFoundContent;
     }
-    return renderEmpty?.('Select') || <DefaultRenderEmpty componentName="Select" />;
+    return renderEmpty?.('Select') || <DefaultRenderEmpty componentName='Select' />;
```
Comment (li-jia-nan) [inline components/mentions/index.tsx:132]:
```suggestion
    return renderEmpty?.('Select') || <DefaultRenderEmpty componentName="Select" />;
```

File: components/mentions/index.tsx [hunk 7]
```diff
@@ -127,14 +129,14 @@ const InternalMentions: React.ForwardRefRenderFunction<MentionsRef, MentionProps
...
   }, [notFoundContent, renderEmpty]);
 
   const mentionOptions = React.useMemo<React.ReactNode>(() => {
     if (loading) {
       return (
-        <Option value="ANTD_SEARCHING" disabled>
-          <Spin size="small" />
+        <Option value='ANTD_SEARCHING' disabled>
```
Comment (li-jia-nan) [inline components/mentions/index.tsx:138]:
```suggestion
        <Option value="ANTD_SEARCHING" disabled>
```

File: components/mentions/index.tsx [hunk 8]
```diff
@@ -134,7 +136,7 @@ const InternalMentions: React.ForwardRefRenderFunction<MentionsRef, MentionProps
     if (loading) {
       return (
         <Option value="ANTD_SEARCHING" disabled>
-          <Spin size="small" />
+          <Spin size='small' />
```
Comment (li-jia-nan) [inline components/mentions/index.tsx:139]:
```suggestion
          <Spin size="small" />
```

File: components/mentions/index.tsx [hunk 9]
```diff
@@ -146,7 +148,7 @@ const InternalMentions: React.ForwardRefRenderFunction<MentionsRef, MentionProps
         {
           value: 'ANTD_SEARCHING',
           disabled: true,
-          label: <Spin size="small" />,
+          label: <Spin size='small' />,
```
Comment (li-jia-nan) [inline components/mentions/index.tsx:151]:
```suggestion
          label: <Spin size="small" />,
```

File: components/qrcode/index.tsx [hunk 10]
```diff
@@ -85,7 +87,7 @@ const QRCode: React.FC<QRCodeProps> = (props) => {
             <>
               <p className={`${prefixCls}-expired`}>{locale?.expired}</p>
               {onRefresh && (
-                <Button type="link" icon={<ReloadOutlined />} onClick={onRefresh}>
+                <Button type='link' icon={<ReloadOutlined />} onClick={onRefresh}>
```
Comment (li-jia-nan) [inline components/qrcode/index.tsx:90]:
```suggestion
                <Button type="link" icon={<ReloadOutlined />} onClick={onRefresh}>
```

File: components/spin/index.tsx [hunk 11]
```diff
@@ -144,7 +146,7 @@ const Spin: React.FC<SpinClassProps> = (props) => {
       {...divProps}
       style={mergedStyle}
       className={spinClassName}
-      aria-live="polite"
+      aria-live='polite'
```
Comment (li-jia-nan) [inline components/spin/index.tsx:149]:
```suggestion
      aria-live="polite"
```

File: components/spin/index.tsx [hunk 12]
```diff
@@ -158,8 +160,8 @@ const Spin: React.FC<SpinClassProps> = (props) => {
         {...divProps}
         className={classNames(`${prefixCls}-nested-loading`, wrapperClassName, hashId)}
       >
-        {spinning && <div key="loading">{spinElement}</div>}
-        <div className={containerClassName} key="container">
+        {spinning && <div key='loading'>{spinElement}</div>}
```
Comment (li-jia-nan) [inline components/spin/index.tsx:163]:
```suggestion
        {spinning && <div key="loading">{spinElement}</div>}
```

File: components/select/index.tsx [hunk 13]
```diff
@@ -152,7 +154,7 @@ const InternalSelect = <
   } else if (mode === 'combobox') {
     mergedNotFound = null;
   } else {
-    mergedNotFound = renderEmpty?.('Select') || <DefaultRenderEmpty componentName="Select" />;
+    mergedNotFound = renderEmpty?.('Select') || <DefaultRenderEmpty componentName='Select' />;
```
Comment (li-jia-nan) [inline components/select/index.tsx:157]:
```suggestion
    mergedNotFound = renderEmpty?.('Select') || <DefaultRenderEmpty componentName="Select" />;
```

File: components/spin/index.tsx [hunk 14]
```diff
@@ -159,7 +161,7 @@ const Spin: React.FC<SpinClassProps> = (props) => {
         className={classNames(`${prefixCls}-nested-loading`, wrapperClassName, hashId)}
       >
         {spinning && <div key="loading">{spinElement}</div>}
-        <div className={containerClassName} key="container">
+        <div className={containerClassName} key='container'>
```
Comment (li-jia-nan) [inline components/spin/index.tsx:164]:
```suggestion
        <div className={containerClassName} key="container">
```

File: components/steps/index.tsx [hunk 15]
```diff
@@ -111,7 +113,7 @@ const Steps: CompoundedComponent = (props) => {
       return (
         <div className={`${prefixCls}-progress-icon`}>
           <Progress
-            type="circle"
+            type='circle'
```
Comment (li-jia-nan) [inline components/steps/index.tsx:116]:
```suggestion
            type="circle"
```

File: components/time-picker/index.tsx [hunk 16]
```diff
@@ -18,7 +20,7 @@ export interface TimeRangePickerProps extends Omit<RangePickerTimeProps<Dayjs>,
 }
 
 const RangePicker = React.forwardRef<any, TimeRangePickerProps>((props, ref) => (
-  <InternalRangePicker {...props} picker="time" mode={undefined} ref={ref} />
+  <InternalRangePicker {...props} picker='time' mode={undefined} ref={ref} />
```
Comment (li-jia-nan) [inline components/time-picker/index.tsx:23]:
```suggestion
  <InternalRangePicker {...props} picker="time" mode={undefined} ref={ref} />
```

File: components/transfer/index.tsx [hunk 17]
```diff
@@ -319,7 +321,7 @@ const Transfer = <RecordType extends TransferItem = TransferItem>(
 
   const getLocale = (transferLocale: TransferLocale) => ({
     ...transferLocale,
-    notFoundContent: renderEmpty?.('Transfer') || <DefaultRenderEmpty componentName="Transfer" />,
+    notFoundContent: renderEmpty?.('Transfer') || <DefaultRenderEmpty componentName='Transfer' />,
```
Comment (li-jia-nan) [inline components/transfer/index.tsx:324]:
```suggestion
    notFoundContent: renderEmpty?.('Transfer') || <DefaultRenderEmpty componentName="Transfer" />,
```

File: components/tree-select/index.tsx [hunk 18]
```diff
@@ -184,7 +186,7 @@ const InternalTreeSelect = <
   if (notFoundContent !== undefined) {
     mergedNotFound = notFoundContent;
   } else {
-    mergedNotFound = renderEmpty?.('Select') || <DefaultRenderEmpty componentName="Select" />;
+    mergedNotFound = renderEmpty?.('Select') || <DefaultRenderEmpty componentName='Select' />;
```
Comment (li-jia-nan) [inline components/tree-select/index.tsx:189]:
```suggestion
    mergedNotFound = renderEmpty?.('Select') || <DefaultRenderEmpty componentName="Select" />;
```

**Comment by li-jia-nan** [comment]
> html tag 都用双引号，要还原下

@zombieJ @MadCcc done

**Comment by MadCcc** [comment]
changelog 改改，应该是“支持 Nextjs app router”

**Comment by li-jia-nan** [comment]
> changelog 改改，应该是“支持 Nextjs app router”

done

**Review by MadCcc — APPROVED** [review]

**Pull request merged**